File: syncit/assets/js/user_socket.js
import {Socket} from "phoenix"

let socket = new Socket("/socket")
socket.connect()

let channel = socket.channel("lobby", {})
channel.join()
    .receive("ok", resp => {
        console.log("Joined successfully", resp)
    })
    .receive("error", resp => {
        console.log("Unable to join", resp)
    })
channel.on("play", payload => {
    let timingDiff = player.getCurrentTime() - payload.currentTime;
    player.playVideo();
    if (timingDiff >= 2.0 || timingDiff <= -2.0) {
        console.debug("seekTo")
        player.seekTo(payload.currentTime)
    }
})
channel.on("pause", () => {
    player.pauseVideo()
})

window.roomChannel = channel

export default socket

File: syncit/lib/syncit_web/channels/user_socket.ex
defmodule SyncitWeb.UserSocket do
  use Phoenix.Socket
  channel "lobby", SyncitWeb.LobbyChannel

  @impl true
  def connect(_params, socket, _connect_info) do
    {:ok, socket}
  end

  @impl true
  def id(_socket), do: nil
end
